### PR TITLE
Opened up DicomAnonymizer so it can be more easily used as a base class

### PR DESCRIPTION
**Pull request by MichaelM223**
This PR is an enhancement of the DicomAnonymizer class to make it easier and safer to use as a base class.

We're using the DicomAnonymizer for multiple scenarios for which we have some custom implementations of anonymizing for example the UID's. Because all internals of the DicomAnonymizer class are private, we have to copy/paste private methods and fields from the base class when we want to override a public method but keep the internals identical.
Instance methods could also only be hidden by subclasses; I've added the `virtual` modifier to allow proper overriding as well.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
~~- [ ] I have updated API documentation~~
~~- [ ] I have included unit tests~~
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Changed all `private` modifiers of `DicomAnonymizer` and `SecurityProfile` to `protected` so they can be used in subclasses
- Added the `virtual` modifier to instance methods of `DicomAnonymizer` so they can be properly overridden

### CODE REVIEW

File: FO-DICOM.Core/DicomAnonymizerGenerated.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public partial class SecurityProfile
             /// - CleanStructdContOption
             /// - CleanGraphOption
             /// </remarks>
-            private const string DefaultProfile = @"
```
Comment (gofal) [inline FO-DICOM.Core/DicomAnonymizerGenerated.cs:30]: This is some generated code. you also have to change the T4 file DicomAnonymizerGenerated.tt in line 31

Comment (MichaelM223) [reply FO-DICOM.Core/DicomAnonymizerGenerated.cs:30]: Ah, overlooked that one. I'll commit the change in a few minutes

**Comment by MichaelM223** [comment]
Fixed the T4 template

**Review by gofal — APPROVED** [review]

File: ChangeLog5.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 * Ignore empty VOI LUT and Modality LUT Sequence (#1369)
 * Validate calling AE and called AE length when creating a DicomClient (#1323)
 * Improve handling of WSI creation: faster offset table calucation and a naming of temp files to allow more than 64.000.
+* Change: DicomAnonymizer private fields and methods changed to protected so they can be used in subclasses, made instance methods virtual so the can be overridden in subclasses
```
Comment (mrbean-bremen) [inline ChangeLog5.md]: typo: the -> they

**Review by mrbean-bremen — APPROVED** [review]

**Comment by MichaelM223** [comment]
Fixed typo in the changelog

File: ChangeLog5.md (hunk 2 above)
Comment (MichaelM223) [reply ChangeLog5.md]: Fixed the typo with [ba65ed3](https://github.com/fo-dicom/fo-dicom/pull/1375/commits/ba65ed3c3b3749b31ab0711737629d978005666b)

File: FO-DICOM.Core/DicomAnonymizer.cs [hunk 3]
```diff
@@ -247,7 +247,7 @@ public void AnonymizeInPlace(DicomDataset dataset)
 
         /// <summary>Anonymizes the dataset of an existing Dicom file</summary>
         /// <param name="file">The file containing the dataset to be altered</param>
-        public void AnonymizeInPlace(DicomFile file)
+        public virtual void AnonymizeInPlace(DicomFile file)
         {
             AnonymizeInPlace(file.Dataset);
             if (file.FileMetaInfo != null)
```
Comment (MichaelM223) [inline FO-DICOM.Core/DicomAnonymizer.cs:253]: Found an issue not related to this PR;
FileMetaInfo does not get parsed by the anonymizer, by adding the line `AnonymizeInPlace(file.FileMetaInfo);` after `if (file.FileMetaInfo != null)` and before `file.FileMetaInfo.MediaStorageSOPInstanceUID = file.Dataset.GetSingleValue<DicomUID>(DicomTag.SOPInstanceUID);`, the meta info should be correctly anonymized (if needed) as well.

Should I add the fix in this PR or create a new one?

**Pull request merged**